source/docs/software/actuators/programming-leds.rst

```diff
@@ -0,0 +1,147 @@
+Programming LEDs
+================
+
+LED strips have been commonly used by teams for several years for a variety of reasons. They allow teams to debug robot functionality from the audience, provide a visual marker for their robot, and can simply add some visual appeal. WPILib has an API for controlling WS2812 LEDs with their data pin connected via PWM.
+
+Instantiating the AddressableLED Object
+---------------------------------------
+
+You first create an ``AddressableLED`` object that takes the PWM port as an argument. It *must* be a PWM header on the roboRIO. Then you set the number of LEDs located on your LED strip, with can be done with the ``setLength()`` function.
+
+.. important:: It is important to note that setting the length of the LED header is an expensive task and it's **not** recommended to run this periodically.
+
+After the length of the strip has been set, you'll have to create an ``AddressableLEDBuffer`` object that takes the number of LEDs as an input. You'll then call ``myAddressableLed.setData(myAddressableLEDBuffer)`` to set the led output data. Finally, you can call ``myAddressableLed.start()`` to write the output continuously. Below is a full example of the initialization process.
+
+.. note:: C++ does not have an AddressableLEDBuffer, and instead uses an Array.
+
+.. tabs::
+
+   .. group-tab:: Java
+
+      .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/master/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/addressableled/Robot.java
+         :language: java
+         :lines: 8-23
+         :linenos:
+         :lineno-start: 14
+
+   .. group-tab:: C++
+
+      .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/master/wpilibcExamples/src/main/cpp/examples/AddressableLED/cpp/Robot.cpp
+         :language: cpp
+         :lines: 14-23
+         :linenos:
+         :lineno-start: 14
+
+      .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/master/wpilibcExamples/src/main/cpp/examples/AddressableLED/cpp/Robot.cpp
+         :language: cpp
+         :lines: 41-47
+         :linenos:
+         :lineno-start: 41
+
+Setting the Entire Strip to One Color
+-------------------------------------
+
+Color can be set to an individual led on the strip using two methods.``setHSV()`` which takes HSV values as an input, and ``setRGB`` which takes RGB values as an input.
+
+Using HSV Values
+^^^^^^^^^^^^^^^^
+
+HSV stands for Hue, Saturation, and Value. Hue describes the color or tint, saturation being the amount of gray, and value being the brightness. Hue is an integer from 1 - 360. Saturation and Value are integers from 0 - 100. LEDs can be set with the ``setHSV`` method that takes 4 arguments: index of the LED, hue, saturation, and value. An example is shown below for setting the color of an LED strip to red.
```

Comment:
Should define red as 0. A picture such as https://commons.wikimedia.org/wiki/File:HSV_color_solid_cylinder_saturation_gray.png#/media/File:Hsl-hsv_models_b.svg is worth a thousand words